src/nova-components/CreateFavorites/resources/js/components/Tool.vue

```diff
@@ -3,49 +3,117 @@
-          :input="inputAuthor"
-        />
-        <text-field
-          :isRequired="true"
-          :type="urlInput"
-          :label="source"
-          :placeholder="source"
-          :error="errorSource"
-          :validationError="validationErrorSource"
-          :changeInput="changeInput"
-          :input="inputSource"
-        />
-        <text-field
-          :isRequired="true"
-          :type="dateInput"
-          :label="postDate"
-          :placeholder="postDate"
-          :error="errorPostDate"
-          :changeInput="changeInput"
-          :input="inputPostDate"
-        />
-        <dropdown-field
-          :label="user"
-          :error="errorUser"
-          :changeInput="changeInput"
-          :input="inputUser"
-        />
+        <div class="field-container">
+          <div class="label-spacing">
+            <label-text :label="title" :isRequired="true" />
+          </div>
+          <div class="input-spacing">
+            <text-input
+              :type="textInput"
+              :error="errorTitle"
+              :sameError="errorSameTitle"
+              :placeholder="title"
+              :changeInput="changeInput"
+              :input="inputTitle"
+            />
+            <error-label
+              v-if="errorTitle"
+              :label="title"
+              :errorType="emptyError"
+            />
+            <error-label
+              v-else-if="errorSameTitle"
+              :label="title"
+              :errorType="sameError"
+            />
+          </div>
+        </div>
+
+        <div class="field-container">
+          <div class="label-spacing">
+            <label-text :label="author" :isRequired="false" />
+          </div>
+          <div class="input-spacing">
+            <text-input
+              :type="textInput"
+              :placeholder="author"
+              :changeInput="changeInput"
+              :input="inputAuthor"
+            />
+          </div>
+        </div>
+
+        <div class="field-container">
+          <div class="label-spacing">
+            <label-text :label="source" :isRequired="true" />
+          </div>
+          <div class="input-spacing">
+            <text-input
+              :type="urlInput"
+              :error="errorSource"
+              :validationError="validationErrorSource"
+              :placeholder="source"
+              :changeInput="changeInput"
+              :input="inputSource"
+            />
+            <error-label
+              v-if="errorSource && validationErrorSource"
+              :label="source"
+              :errorType="emptyAndInvalidError"
+            />
+            <error-label
+              v-else-if="errorSource"
+              :label="source"
+              :errorType="emptyError"
+            />
+            <error-label
+              v-else-if="validationErrorSource"
+              :label="source"
+              :errorType="validationError"
+            />
```

Comment:
don't use multiple error-label nodes, find a way to do it with computed attr and one node